docs/src/whatsnew/latest.rst

```diff
@@ -166,6 +166,10 @@ This document explains the changes made to Iris for this release
 #. `@wjbenfold`_ improved the speed of linear interpolation using
    :meth:`iris.analysis.trajectory.interpolate` (:pull:`4366`)
 
+#. NumPy ``v1.23`` behaviour changes mean that
+   :func:`iris.experimental.ugrid.utils.recombine_submeshes` now uses ~3x as
+   much memory - up to 700MB for the largest expected data chunks. Investigated
```

Comment:
I don't think we can go into the details of how you might calculate this.
But also we can't really justify  "largest expected data" for general usage.
So, can we just indicate the rough size of a testcase that takes X memory?

Reply:
4ffc94ddccf3dea8375bc51a73584bfe1d7bd646

Reply:
:+1: sorted !